### DISCUSSION TITLE
Wake resolution Settings

### DISCUSSION THREAD

**Discussion by Zehai-Huang** [NatLabRockies/floris#772] [Q&A]
Hi,
    How should the parameters be set to achieve the level of precision shown in Figure 1? Figure 1 is what I saved in the document (https://nrel.github.io/floris/index.html#). I tried to change the parameters **x_resolution** and **y_resolution**, but there was no significant change.(Figure 2 is my own simulation)
figure1:
![figure1](https://github.com/NREL/floris/assets/79148481/63d76760-4155-4ade-b3e1-0f3f9d2bfdac)
figure2:
![figure2](https://github.com/NREL/floris/assets/79148481/7831ac94-08f6-414b-8620-77c0cd184a42)

Sincerely,

**Comment by Zehai-Huang** [comment]
my code:

`fi = FlorisInterface("inputs/gch.yaml")`
`horizontal_plane = fi.calculate_horizontal_plane(
    x_resolution=2000,
    y_resolution=1000,
    height=90.0,
    yaw_angles=np.array([[[0.,0.,0]]]),
)`
`wakeviz.visualize_cut_plane(
    horizontal_plane,
    # ax=ax_list[0],
    label_contours=True,
    title="Horizontal"
)`

**Comment by misi9170** [comment]
Hi @Zehai-Huang , you could try changing the `clevels` keyword argument to `visualize_cut_plane()`. 

For example, setting `clevels=np.linspace(0, 10, 100)` will provide 100 color levels ranging between 0 and 10 m/s. The following code
```
import numpy as np
import matplotlib.pyplot as plt

from floris.tools import FlorisInterface
import floris.tools.visualization as wakeviz

fi = FlorisInterface("inputs/gch.yaml")
horizontal_plane = fi.calculate_horizontal_plane(height=90.0, yaw_angles=np.array([[[0.,0.,0]]]))

fig, ax = plt.subplots(2, 1)
wakeviz.visualize_cut_plane(horizontal_plane, clevels=np.linspace(0, 10, 100), ax=ax[0])
wakeviz.visualize_cut_plane(horizontal_plane, clevels=np.linspace(0, 10, 100), levels=[], ax=ax[1])
ax[0].set_title("With contour lines")
ax[1].set_title("Without contour lines")

plt.show()
```
produces
![772](https://github.com/NREL/floris/assets/39596329/d61d959d-5ea1-4a19-a194-87dee42ca66f)

Is that what you were after?

**Reply by Zehai-Huang** [reply]
I already know how to modify it. Thank you very much for your answer！

**Reply by Nellytado** [reply]
Hi, 

How can I remove the labels off the contour lines which can be seen in FLORIS vs. I used the code however labels on the contour lines were present even if i added 'contour_labels=False' to my line of code. 

Thank you!

**Reply by misi9170** [reply]
Hi @Nellytado , I'm not sure that I understand your question. Using FLORIS v4, the following code
```python
import numpy as np
import matplotlib.pyplot as plt

from floris import FlorisModel
import floris.flow_visualization as flowviz

fi = FlorisModel("inputs/gch.yaml")
fi.run()
horizontal_plane = fi.calculate_horizontal_plane(height=90.0)

fig, ax = plt.subplots(2, 1)
flowviz.visualize_cut_plane(horizontal_plane, clevels=np.linspace(0, 10, 100), ax=ax[0], label_contours=False)
flowviz.visualize_cut_plane(horizontal_plane, clevels=np.linspace(0, 10, 100), ax=ax[1], label_contours=True)
ax[0].set_title("Without contour labels")
ax[1].set_title("With contour labels")

fig.savefig("contour_labels.png")

plt.show()
```
produces
![contour_labels](https://github.com/NREL/floris/assets/39596329/7367b988-b3fe-40bb-803a-03563a624d65)

Here, the `label_contours` flag seems to be operating correctly. If you mean that you would like to remove the contour lines, this is achieved by setting `levels=[]`, as shown in my original response.

**Reply by Nellytado** [reply]
Hi, 
but on Floris V3 it does not let me remove the contour labels. This is my code to generate the wakes of a 6 turbine farm on FLORIS (image attached). I want to keep the contour lines but remove the velcoity labels on them. I also want to add a legend to the max inflow speed I set. As you can see the legend goes all the way till 9m/s instead of 8m/s and the contour labels can still be seen.
![image](https://github.com/NREL/floris/assets/165584201/8426c5ae-e1fe-4edd-8bd5-632913ae745f)

**Reply by misi9170** [reply]
The ability to remove the contour labels was introduced to FLORIS v3 in October 2023, see #719, and is in FLORIS version 3.5 onward. If you are using a version of FLORIS earlier than 3.5, the option will not be available unfortunately (although it may be straightforward for you to upgrade your FLORIS to version 3.5, or to 3.6, the final version before FLORIS v4.0).

On the maximum velocity in the colorbar---I tried to get that to work as you wanted, but unfortunately I also was not immediately able to set the maximum velocity. It may be possible to get the colorbar from the plot and alter it directly using [matplotlib](https://matplotlib.org/). However, it would be nice to have an easy way to set the limits for the colorbar in the `visualize_cut_plane` method. Feel free to [open an issue](https://github.com/nrel/floris/issues) on that, and we'd certainly be interested to hear if you find a solution yourself.
